### PR TITLE
Implement adding/editing blog post in yii-demo

### PR DESCRIPTION
**Pull request by Fantom409**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

File: src/Blog/Post/PostController.php [hunk 1]
```diff
@@ -2,28 +2,171 @@
...
+use Yiisoft\Http\Method;
+use Yiisoft\Router\UrlGeneratorInterface;
+use Yiisoft\Yii\Web\User\User as UserComponent;
 
 final class PostController
 {
     private ViewRenderer $viewRenderer;
+    private ResponseFactoryInterface $responseFactory;
+    private LoggerInterface $logger;
+    private UserComponent $userComponent;
```
Comment (samdark) [inline src/Blog/Post/PostController.php:24]:
```suggestion
    private UserComponent $userService;
```

File: views/blog/post/__form.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    echo Alert::widget()
+        ->options(['class' => 'alert-danger'])
+        ->body(
+            Html::encode($error)
+        );
+}
+?>
+
+
+<h1><?= $title ?></h1>
```
Comment (samdark) [inline views/blog/post/__form.php:24]:
```suggestion
<h1><?= Html::encode($title) ?></h1>
```

File: src/Blog/Post/PostController.php [hunk 3]
```diff
@@ -2,28 +2,171 @@
...
+        $parameters = [
+            'body' => $body,
+            'action' => ['blog/add']
+        ];
+
+        if ($request->getMethod() === Method::POST) {
+            $error = '';
+
+            try {
+                foreach (['header', 'content'] as $name) {
```
Comment (samdark) [inline src/Blog/Post/PostController.php:71]: It is called "title".

```suggestion
                foreach (['title', 'content'] as $name) {
```

File: views/blog/post/__form.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+<form id="postForm"
+      method="POST"
+      action="<?= $urlGenerator->generate(...$action) ?>"
+      enctype="multipart/form-data"
+>
+    <input type="hidden" name="_csrf" value="<?= $csrf ?>">
+    <div class="form-group">
+        <label for="header">Header</label>
```
Comment (samdark) [inline views/blog/post/__form.php:33]:
```suggestion
        <label for="title">Title</label>
```

**Review by samdark — CHANGES_REQUESTED** [review]

**Review by samdark — APPROVED** [review]

File: src/Command/User/CreateCommand.php [hunk 5]
```diff
@@ -9,18 +9,24 @@
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Style\SymfonyStyle;
+use Yiisoft\Rbac\Manager;
+use Yiisoft\Rbac\StorageInterface;
 use Yiisoft\Yii\Console\ExitCode;
 use Yiisoft\Yii\Cycle\Command\CycleDependencyProxy;
 
 class CreateCommand extends Command
```
Comment (samdark) [inline src/Command/User/CreateCommand.php:17]: The command currently fails with:

image

**Comment by samdark** [comment]
It would be great to add a button that leads from post full view to editing a post.

File: src/Command/User/CreateCommand.php (hunk 5 above)
Comment (samdark) [reply src/Command/User/CreateCommand.php:17]: A command to set RBAC role for existing user would be handy.

Comment (Fantom409) [reply src/Command/User/CreateCommand.php:17]: > The command currently fails with:
> 
> image

I couldn't reproduce  error. Most likely you have old db schema

**Comment by samdark** [comment]
Thanks!

**Comment by samdark** [comment]
btw., how about adding tags adding / editing as well?

**Comment by Fantom409** [comment]
> btw., how about adding tags adding / editing as well?

https://trello.com/c/xkJxzzw2/113-implement-adding-editing-tags-in-yii-demo

**Pull request merged**